### PR TITLE
[IMP] Function: add `hidden` tag to hide functions

### PR DESCRIPTION
**Pull request by yunchen4**
## Description:

This PR adds the tag `hidden` to hide some functions to users, so users will not have the chance to see their existences.

Odoo task ID : [3102309](https://www.odoo.com/web#id=3102309&action=333&active_id=2328&model=project.task&view_type=form&cids=1&menu_id=4720)

## review checklist

- [ ] feature is organized in plugin, or UI components
- [ ] support of duplicate sheet (deep copy)
- [ ] in model/core: ranges are Range object, and can be adapted (adaptRanges)
- [ ] in model/UI: ranges are strings (to show the user)
- [ ] undo-able commands (uses this.history.update)
- [ ] multiuser-able commands (has inverse commands and transformations where needed)
- [ ] new/updated/removed commands are documented
- [ ] exportable in excel
- [ ] translations (\_lt("qmsdf %s", abc))
- [ ] unit tested
- [ ] clean commented code
- [ ] track breaking changes
- [ ] doc is rebuild (npm run doc)
- [ ] status is correct in Odoo

### CODE REVIEW

File: src/components/composer/composer/composer.ts [hunk 1]
```diff
@@ -169,6 +167,17 @@ export class Composer extends Component<Props, SpreadsheetChildEnv> {
...
+    return Object.keys(functionRegistry.content)
+      .filter((key) => {
+        return !functionRegistry.content[key].hidden;
+      })
+      .reduce((prevContent, currentKey) => {
+        prevContent[currentKey] = functionRegistry.content[currentKey];
+        return prevContent;
+      }, {});
+  }
+
```
Comment (yunchen4) [inline src/components/composer/composer/composer.ts:180]: Should I convert it into a state and use `useEffect` to update the state? Using `useEffect` seems to have less calculation.

File: src/components/composer/composer/composer.ts [hunk 2]
```diff
@@ -370,7 +379,7 @@ export class Composer extends Component<Props, SpreadsheetChildEnv> {
 
   showAutocomplete(searchTerm: string) {
     this.autoCompleteState.showProvider = true;
-    let values = Object.entries(functionRegistry.content).map(([text, { description }]) => {
+    let values = Object.entries(this.nonHiddenFunctions).map(([text, { description }]) => {
```
Comment (LucasLefevre) [inline src/components/composer/composer/composer.ts:382]: what about just adding `.filter(([text, { hidden }]) => !hidden)` ?
It look simpler than unpack, filter, reduce
```suggestion
    let values = Object.entries(functionRegistry.content).filter(([text, { hidden }]) => !hidden).map(([text, { description }]) => {
```

File: src/components/composer/composer/composer.ts (hunk 1 above)
Comment (LucasLefevre) [reply src/components/composer/composer/composer.ts:180]: in practice I don't think it is a performance bottleneck to filter the functions every time.
There only a few hundred functions. Iterating that amount of items should be very fast

File: src/types/functions.ts [hunk 3]
```diff
@@ -36,13 +36,15 @@ export interface AddFunctionDescription {
...
   isExported?: boolean;
+  hidden?: boolean;
 }
 
 export interface FunctionDescription extends AddFunctionDescription {
   minArgRequired: number;
   maxArgPossible: number;
   nbrArgRepeating: number;
   getArgToFocus: (argPosition: number) => number;
+  hidden?: boolean;
```
Comment (LucasLefevre) [inline src/types/functions.ts]: not useful since it extends `AddFunctionDescription` which has the same property definition

**Review by LucasLefevre — APPROVED** [review]
LGTM
(I applied my comment)
Side note about commit messages: the line length shouldn't be above ~80 characters :wink: 
robodoo r+

**Pull request closed without merging**